src/Lexer.php

```diff
@@ -21,7 +21,18 @@
  *
  * @author Konstantin Kudryashov <ever.zet@gmail.com>
  *
- * @phpstan-type TToken array{type: string, line: int, value: string|null, deferred: bool, tags?: list<string>, keyword?: string, keyword_type?: string, text?: string, indent?: int, columns?: list<string>}
+ * @phpstan-import-type TKeywordsString from KeywordsInterface
+ * @phpstan-import-type TKeywordsType from KeywordsInterface
+ * @phpstan-import-type TStepKeywordsType from KeywordsInterface
+ * @phpstan-import-type TGeneralKeywordsType from KeywordsInterface
+ *
+ * @phpstan-type TTokenType 'Text'|'Comment'|'EOS'|'Newline'|'PyStringOp'|'TableRow'|'Tag'|'Language'|TGeneralKeywordsType
+ * @phpstan-type TTokenValue int|string|null
+ * @phpstan-type TToken array{type: TTokenType, value: TTokenValue, line: int, deferred: bool}
+ * @phpstan-type TKeywordToken array{type: TTokenType, value: TTokenValue, line: int, deferred: bool, keyword: string, indent: int}
+ * @phpstan-type TStepToken array{type: TTokenType, value: TTokenValue, line: int, deferred: bool, keyword_type: string, text: string}
+ * @phpstan-type TTagToken array{type: TTokenType, value: TTokenValue, line: int, deferred: bool, tags: list<string>}
```

Comment:
the value or a tag token is always a string. Same for the step token.

If we define more precise types for the different tokens, we could as well do it fully (replacing all the `\assert(\is_string($token['value']))` I added in #342 for such cases.

Reply:
Yep, I fixed those others. 👍